modules/@apostrophecms/login/index.js

```diff
@@ -94,31 +110,52 @@ module.exports = {
         async login(req) {
           const username = self.apos.launder.string(req.body.username);
           const password = self.apos.launder.string(req.body.password);
+          const session = self.apos.launder.boolean(req.body.session);
           if (!(username && password)) {
             throw self.apos.error('invalid');
           }
           const user = await self.apos.login.verifyLogin(username, password);
           if (!user) {
             throw self.apos.error('invalid');
           }
-          const passportLogin = (user) => {
-            return require('util').promisify(function(user, callback) {
-              return req.login(user, callback);
-            })(user);
-          };
-          await passportLogin(user);
+          if (session) {
+            const passportLogin = (user) => {
+              return require('util').promisify(function(user, callback) {
```

Comment:
You lost me here 😅 
I see it was already written like that previously, but still, I am not sure of what is going on here.

Reply:
We do not use callbacks, but passport's login method takes a callback. So we "promisify" it, which allows us to await it. This is the correct pattern if you just invoke something callback-based in A3.